Packages/vcs/Lib/vtk_ui/text.py

```diff
@@ -87,45 +186,146 @@ def __init__(self, interactor, string, movable=False, on_move=None, on_drag=None
         else:
             self.actor = text_actor(string, fgcolor, size, font)
 
-        widget.SetTextActor(self.actor)
+        widget = WidgetReprShim(interactor, self.actor)
```

Comment:
I discovered a few issues with using the `vtkTextWidget` and `vtkTextRepresentation`. The biggest one was buried in the description of [this property– UseBorderAlign](http://www.vtk.org/doc/nightly/html/classvtkTextActor.html#a091d92e16bd71a19b02334bef839038f). I couldn't accurately predict the position or rotated text, which was necessary for a number of pieces of interaction (The cursor for the TextBox, clicking to place the cursor, accurately placing dragged text...). Instead of fighting against the Widget/Representation, I implemented a simple shim object that does all of the things that I was personally using the Widget/Representation for, by directly acting on the `vtkTextActor`. This let me keep most of the code for managing the Label widget the same, while dramatically changing the guts of how it handles all sorts of things.

TL;DR: I use `vtkTextActor` directly instead of using `vtkTextRepresentation` and `vtkTextWidget` to manager a `vtkTextActor` for me, because I was having trouble doing it the other way.